buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/PrecommitTasks.groovy

```diff
@@ -144,18 +144,19 @@ class PrecommitTasks {
         project.tasks.withType(CheckForbiddenApis).configureEach {
             dependsOn(buildResources)
 
-            //  use the runtime classpath if we have it, but some qa projects don't have one...
-            if (name.endsWith('Test')) {
-                FileCollection runtime = project.sourceSets.test.runtimeClasspath
-                if (runtime != null) {
-                    classpath = runtime.plus(project.sourceSets.test.compileClasspath)
-                }
-            } else {
-                FileCollection runtime = project.sourceSets.main.runtimeClasspath
-                if (runtime != null) {
-                    classpath = runtime.plus(project.sourceSets.main.compileClasspath)
-                }
+            //parse out the sourceSetName
+            String[] parts = name.split(ForbiddenApisPlugin.FORBIDDEN_APIS_TASK_NAME)
```

Comment:
Instead of a split, could we assert the name startsWith and then do a substring using the length? Otherwise we have an edge case (part.length != 2) that should never happen that we allow for errors in this logic.

Reply:
Currently the main task name is `forbiddenApisMain`, however, if the plugin drops the `Main` suffix such that task name is `forbiddenApis` (which iiuc is the standard naming convention [1]), the code as-is should continue work. For this reason, I would prefer to keep it as-is.

[1]
https://docs.gradle.org/current/userguide/building_java_projects.html#sec:java_source_sets
```
Most language plugins, Java included, automatically create a source set called main, which is used for the project’s production code. This source set is special in that its name is not included in the names of the configurations and tasks, hence why you have just a compileJava task and compileOnly and implementation configurations rather than compileMainJava, mainCompileOnly and mainImplementation respectively.
```

Reply:
It's not that I don't think the code as is would work, it's that by splitting the code is more difficult to understand since at a glance it looks like we are ignoring `parts[0]`, which is actually always an empty string.

Reply:
Switched to use substring in https://github.com/elastic/elasticsearch/pull/54731/commits/309c7a92955fb8f1bf038362df4bbe24d1cfe97e